pkg/trust/trust.go

```diff
@@ -0,0 +1,250 @@
+	mergedConfig := RegistryConfiguration{Docker: map[string]RegistryNamespace{}}
+	dockerDefaultMergedFrom := ""
+	nsMergedFrom := map[string]string{}
+
+	dir, err := os.Open(dirPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return &mergedConfig, nil
+		}
+		return nil, err
+	}
+	configNames, err := dir.Readdirnames(0)
+	if err != nil {
+		return nil, err
+	}
+	for _, configName := range configNames {
+		if !strings.HasSuffix(configName, ".yaml") {
+			continue
+		}
+		configPath := filepath.Join(dirPath, configName)
+		configBytes, err := ioutil.ReadFile(configPath)
+		if err != nil {
+			return nil, err
+		}
+		var config RegistryConfiguration
+		err = yaml.Unmarshal(configBytes, &config)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Error parsing %s", configPath)
+		}
+		if config.DefaultDocker != nil {
+			if mergedConfig.DefaultDocker != nil {
+				return nil, errors.Errorf(`Error parsing signature storage configuration: "default-docker" defined both in "%s" and "%s"`,
+					dockerDefaultMergedFrom, configPath)
+			}
+			mergedConfig.DefaultDocker = config.DefaultDocker
+			dockerDefaultMergedFrom = configPath
+		}
+		for nsName, nsConfig := range config.Docker { // includes config.Docker == nil
+			if _, ok := mergedConfig.Docker[nsName]; ok {
+				return nil, errors.Errorf(`Error parsing signature storage configuration: "docker" namespace "%s" defined both in "%s" and "%s"`,
+					nsName, nsMergedFrom[nsName], configPath)
+			}
+			mergedConfig.Docker[nsName] = nsConfig
+			nsMergedFrom[nsName] = configPath
+		}
+	}
+	return &mergedConfig, nil
+}
+
+// HaveMatchRegistry checks if trust settings for the registry have been configed in yaml file
+func HaveMatchRegistry(key string, registryConfigs *RegistryConfiguration) *RegistryNamespace {
+	searchKey := key
+	if !strings.Contains(searchKey, "/") {
+		val, exists := registryConfigs.Docker[searchKey]
+		if exists {
+			return &val
+		}
+	}
+	for range strings.Split(key, "/") {
+		val, exists := registryConfigs.Docker[searchKey]
+		if exists {
+			return &val
+		}
+		if strings.Contains(searchKey, "/") {
+			searchKey = searchKey[:strings.LastIndex(searchKey, "/")]
+		}
+	}
+	return nil
+}
+
+// CreateTmpFile creates a temp file under dir and writes the content into it
+func CreateTmpFile(dir, pattern string, content []byte) (string, error) {
+	tmpfile, err := ioutil.TempFile(dir, pattern)
+	if err != nil {
+		return "", err
+	}
+	defer tmpfile.Close()
+
+	if _, err := tmpfile.Write(content); err != nil {
+		return "", err
+
+	}
+	return tmpfile.Name(), nil
+}
+
+// GetGPGId return GPG identity, either bracketed <email> or ID string
+// comma separated if more than one key
+func GetGPGId(keys []string) string {
+	for _, k := range keys {
+		if _, err := os.Stat(k); err != nil {
+			decodeKey, err := base64.StdEncoding.DecodeString(k)
+			if err != nil {
+				logrus.Warnf("error decoding key data")
+				continue
+			}
+			tmpfileName, err := CreateTmpFile("/run/", "", decodeKey)
```

Comment:
`/run` is not writable by non-root. Why not let `ioutil.TempFile` use the default? Or, perhaps, pipe the data into `gpg2` without creating a temporary file at all.